pkg/operator/controller/controller_router_deployment.go

```diff
@@ -78,16 +78,9 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, routerImage strin
 		manifests.OwningClusterIngressLabel: ci.Name,
 	}
 
-	deployment.Spec.Selector = &metav1.LabelSelector{
-		MatchLabels: map[string]string{
-			// control pods which match controller
-			controllerDeploymentLabel: IngressControllerDeploymentLabel(ci),
-		},
-	}
-	deployment.Spec.Template.Labels = map[string]string{
-		// ensure pods are matched to be controlled by the deployment
-		controllerDeploymentLabel: IngressControllerDeploymentLabel(ci),
-	}
+	deployment.Spec.Selector = IngressControllerDeploymentPodSelector(ci)
+
+	deployment.Spec.Template.Labels = IngressControllerDeploymentPodSelector(ci).MatchLabels
```

Comment:
Looks like you could even simplify this to `deployment.Spec.Template.Labels = deployment.Spec.Selector.MatchLabels`.

Reply:
Changed